src/main/java/io/strimzi/kafka/metrics/PrometheusMetricsReporterConfig.java

```diff
@@ -82,4 +126,60 @@ public synchronized Optional<HTTPServer> startHttpServer() {
         }
     }
 
+    static class ListenerValidator implements ConfigDef.Validator {
+
+        @Override
+         public void ensureValid(String name, Object value) {
+            Matcher matcher = Listener.PATTERN.matcher(String.valueOf(value));
+            if (!matcher.matches()) {
+                throw new ConfigException(name, value, "Listener must be of format http://[host]:[port]");
+            }
+        }
+    }
+
+    static class Listener {
+
+        private static final Pattern PATTERN = Pattern.compile("http://\\[?([0-9a-zA-Z\\-%._:]*)]?:([0-9]+)");
+
+        final String host;
+        final int port;
+
+        Listener(String host, int port) {
+            this.host = host;
+            this.port = port;
+        }
+
+        static Listener parseListener(String listener) {
+            Matcher matcher = PATTERN.matcher(listener);
+            if (matcher.matches()) {
+                String host = matcher.group(1);
+                int port = Integer.parseInt(matcher.group(2));
+                return new Listener(host, port);
+            } else {
+                throw new ConfigException(LISTENER_CONFIG, listener, "Listener be of format http://[host]:[port]");
+            }
+        }
+
+        public InetSocketAddress getInetSocketAddress() {
+            return host == null || host.isEmpty() ? new InetSocketAddress(port) : InetSocketAddress.createUnresolved(host, port);
```

Comment:
If we do https://github.com/strimzi/metrics-reporter/pull/16/files#r1504056500 then we can delete this method

Reply:
Method removed